packages/devtools/lib/src/utils.dart

```diff
@@ -151,20 +151,17 @@ bool isLetter(int codeUnit) =>
 ///
 /// Given an input such as
 /// `_WidgetsFlutterBinding&BindingBase&GestureBinding.handleBeginFrame`, this
-/// method will strip off all the Mixin names and return
-/// `_WidgetsFlutterBinding.handleBeginFrame`.
+/// method will strip off all the leading class names and return
+/// `GestureBinding.handleBeginFrame`.
+///
+/// See (https://github.com/dart-lang/sdk/issues/36999).
 String getSimpleStackFrameName(String name) {
-  final firstAmpersandIndex = name.indexOf('&');
-  final firstPeriodIndex = name.indexOf('.');
-
-  if (firstAmpersandIndex != -1 &&
-      firstPeriodIndex != -1 &&
-      firstAmpersandIndex < firstPeriodIndex &&
-      name.length > firstAmpersandIndex + 1) {
-    final nextCharCodeUnit = name[firstAmpersandIndex + 1].codeUnitAt(0);
-    if (isLetter(nextCharCodeUnit) || nextCharCodeUnit == '_'.codeUnitAt(0)) {
-      return name.substring(0, firstAmpersandIndex) +
-          name.substring(firstPeriodIndex);
+  const underscoreCodeUnit = 95;
+  final lastAmpersandIndex = name.lastIndexOf('&');
+  if (lastAmpersandIndex != -1 && name.length > lastAmpersandIndex + 1) {
+    final nextCharCodeUnit = name[lastAmpersandIndex + 1].codeUnitAt(0);
+    if (isLetter(nextCharCodeUnit) || nextCharCodeUnit == underscoreCodeUnit) {
```

Comment:
I think this method can be replaced with the one liner
=> name.split('&').last

Checking for isLetter() or underscore is more harmful than useful as it would be confused by bad class names like `$SomeClassName`
and I can't think of any cases where the check should really help us assuming no bugs in the VM.

Reply:
I believe we discussed this either on the original PR or offline, but we need to check that it is a letter because some of our class names have ampersands that aren't due to appending classes. 

For example: `dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)`. This class name has ampersands that seem to be for c++ references. We want to make sure we don't trim this class name.

Maybe we can check that the following char is not a comma, but that also seems problematic.